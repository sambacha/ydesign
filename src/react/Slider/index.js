import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { StyledInput } from '../Input';
import theme from '../theme';

const StyledSlider = styled(StyledInput)`
  & {
    appearance: none;
  }
  &:hover {
    box-shadow: none;
  }
  &::-webkit-slider-runnable-track {
    box-sizing: border-box;
    appearance: none;
    border: none;
    border-radius: 4px;
    background: currentColor;
    height: 4px;
    display: flex;
    align-items: center;
  }
  &::-webkit-slider-thumb {
    box-sizing: border-box;
    appearance: none;
    border: 1px solid #ccc;
    background: white;
    height: 1.5rem;
    width: 1.5rem;
    border-radius: 100%;
    cursor: grab;
    &:hover {
      box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
    }
    &:active {
      cursor: grabbing;
      border-color: currentColor;
    }
  }

  &::-moz-range-track {
    box-sizing: border-box;
    appearance: none;
    border: none;
    border-radius: 4px;
    background: currentColor;
    height: 4px;
    display: flex;
    align-items: center;
  }
  &::-moz-range-thumb {
    box-sizing: border-box;
    appearance: none;
    border: 1px solid #ccc;
    background: white;
    height: 1.5rem;
    width: 1.5rem;
    border-radius: 100%;
    cursor: grab;
    &:hover {
      box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
    }
    &:active {
      cursor: grabbing;
      border-color: currentColor;
    }
  }
`;

const Slider = React.forwardRef((props, ref) => (
  <StyledSlider ref={ref} {...props} type={'range'} />
));

StyledSlider.defaultProps = {
  color: 'primary',
  bg: 'none',
  height: '2rem',
  minWidth: '200px',
  m: 0,
  p: 0,
  border: 'none',
  boxShadow: 'none',
  theme,
};

Slider.defaultProps = {
  'aria-label': 'range slider',
};

Slider.propTypes = {
  'aria-label': PropTypes.string,
};

Slider.displayName = 'Slider';

export default Slider;
