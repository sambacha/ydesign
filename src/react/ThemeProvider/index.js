import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
