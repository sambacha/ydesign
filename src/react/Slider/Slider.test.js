import Slider from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Slider component sanity', () => {
  it('has name', () => {
    expect(Slider.displayName).toBe('Slider');
  });

  it('matches default snapshot', () => {
    const component = render(<Slider />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Slider />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
