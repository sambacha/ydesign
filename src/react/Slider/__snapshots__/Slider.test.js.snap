// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component sanity matches default snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  color: #4E3FCE;
  background-color: none;
  font-size: 1rem;
  line-height: initial;
  height: 2rem;
  padding: 0;
  border: none;
  border-color: #ccc;
  border-radius: 4px;
  box-shadow: none;
  min-width: 200px;
  margin: 0;
}

.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover {
  box-shadow: 0px 2px 6px rgba(0,0,0,0.3);
}

.c0:focus {
  outline: none;
  border-color: #4E3FCE;
}

.c0:disabled {
  opacity: 0.4;
  pointer-events: none;
}

.c0[type="color"] {
  min-width: 3rem;
  padding: 0;
}

.c0[type="color"]::-webkit-color-swatch-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
}

.c0[type="color"]::-webkit-color-swatch {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 2rem;
  width: 2rem;
  border: none;
  border-radius: 100%;
}

.c0[type="color"]::-moz-color-swatch {
  height: 2rem;
  width: 2rem;
  border: none;
  border-radius: 100%;
}

.was-validated .c0:valid {
  border-color: #28C081;
}

.was-validated .c0:valid ~ svg.icon-valid {
  visibility: visible;
  color: #28C081;
}

.was-validated .c0:invalid {
  border-color: #DC2C10;
}

.was-validated .c0:invalid ~ svg.icon-invalid {
  visibility: visible;
  color: #DC2C10;
}

.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover {
  box-shadow: none;
}

.c0::-webkit-slider-runnable-track {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  border-radius: 4px;
  background: currentColor;
  height: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0::-webkit-slider-thumb {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #ccc;
  background: white;
  height: 1.5rem;
  width: 1.5rem;
  border-radius: 100%;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
}

.c0::-webkit-slider-thumb:hover {
  box-shadow: 0px 2px 4px rgba(0,0,0,0.1);
}

.c0::-webkit-slider-thumb:active {
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
  border-color: currentColor;
}

.c0::-moz-range-track {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  border-radius: 4px;
  background: currentColor;
  height: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0::-moz-range-thumb {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #ccc;
  background: white;
  height: 1.5rem;
  width: 1.5rem;
  border-radius: 100%;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
}

.c0::-moz-range-thumb:hover {
  box-shadow: 0px 2px 4px rgba(0,0,0,0.1);
}

.c0::-moz-range-thumb:active {
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
  border-color: currentColor;
}

<input
  aria-label="range slider"
  class="sc-pIWYI sc-kIPIUN c0"
  color="primary"
  font-size="1rem"
  height="2rem"
  type="range"
/>
`;

exports[`Slider component sanity matches themed snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  color: #4E3FCE;
  background-color: none;
  font-size: 1rem;
  line-height: initial;
  height: 2rem;
  padding: 0;
  border: none;
  border-color: #ccc;
  border-radius: 4px;
  box-shadow: none;
  min-width: 200px;
  margin: 0;
}

.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover {
  box-shadow: 0px 2px 6px rgba(0,0,0,0.3);
}

.c0:focus {
  outline: none;
  border-color: #4E3FCE;
}

.c0:disabled {
  opacity: 0.4;
  pointer-events: none;
}

.c0[type="color"] {
  min-width: 3rem;
  padding: 0;
}

.c0[type="color"]::-webkit-color-swatch-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
}

.c0[type="color"]::-webkit-color-swatch {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 2rem;
  width: 2rem;
  border: none;
  border-radius: 100%;
}

.c0[type="color"]::-moz-color-swatch {
  height: 2rem;
  width: 2rem;
  border: none;
  border-radius: 100%;
}

.was-validated .c0:valid {
  border-color: #28C081;
}

.was-validated .c0:valid ~ svg.icon-valid {
  visibility: visible;
  color: #28C081;
}

.was-validated .c0:invalid {
  border-color: #DC2C10;
}

.was-validated .c0:invalid ~ svg.icon-invalid {
  visibility: visible;
  color: #DC2C10;
}

.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover {
  box-shadow: none;
}

.c0::-webkit-slider-runnable-track {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  border-radius: 4px;
  background: currentColor;
  height: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0::-webkit-slider-thumb {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #ccc;
  background: white;
  height: 1.5rem;
  width: 1.5rem;
  border-radius: 100%;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
}

.c0::-webkit-slider-thumb:hover {
  box-shadow: 0px 2px 4px rgba(0,0,0,0.1);
}

.c0::-webkit-slider-thumb:active {
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
  border-color: currentColor;
}

.c0::-moz-range-track {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  border-radius: 4px;
  background: currentColor;
  height: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0::-moz-range-thumb {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #ccc;
  background: white;
  height: 1.5rem;
  width: 1.5rem;
  border-radius: 100%;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
}

.c0::-moz-range-thumb:hover {
  box-shadow: 0px 2px 4px rgba(0,0,0,0.1);
}

.c0::-moz-range-thumb:active {
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
  border-color: currentColor;
}

<input
  aria-label="range slider"
  class="sc-pIWYI sc-kIPIUN c0"
  color="primary"
  font-size="1rem"
  height="2rem"
  type="range"
/>
`;
